### DISCUSSION TITLE
How can I use the src to build? I've downloaded the source code ,but I want to obtain "spectral-linux" ,this executable file.

### DISCUSSION THREAD

**Discussion by lqywow** [stoplightio/spectral#2166] [Q&A]
I want to know how to get the executable file spectral-linux through the src

**Comment by P0lip** [comment]
Hey!
Assuming you have `yarn` installed, you need to run `yarn build` and then `yarn workspace @stoplight/spectral-cli build.nix`.
The binaries can be found under `packages/cli/binaries` directory.

**Reply by lqywow** [reply]
Ok,thank  U! 
I've downloaded  the source code (zip)](https://github.com/stoplightio/spectral/archive/refs/tags/v6.4.0.zip). I'll try it immediately !!!

**Reply by lqywow** [reply]
Sorry ,I  use the command  #yarn  spectral-src/  @stoplight/spectral-cli build.nix .

There is a error :
Usage  Error : Couldn't find a script named "@stoplight/spectral-cli "

And spectral-src /   —— the directory includes the source code

**Reply by P0lip** [reply]
I think you missed `yarn workspace` in front. The whole command is `yarn workspace @stoplight/spectral-cli build.nix`

**Reply by lqywow** [reply]
sorry , I use The whole command  yarn workspace @stoplight/spectral-cli build.nix  again.
There are some errors:

pkg@5.4.1
&gt; Fetching base Node.js binaries to PKG_CACHE_PATH
  fetched-v14.18.1-linux-loong64      [                    ] 0%
&gt; Error! 404: Not Found
&gt; Not found in remote cache:
  {"tag":"v3.2","name":"node-v14.18.1-linux-loong64"}
&gt; Building base binary from source:
  built-v14.18.1-linux-loong64
&gt; Error! Unknown arch 'loong64'. Specify x64, x86, armv7, arm64, ppc64, s390x

I know the problem is architecture,how can I fix it ?

**Reply by lqywow** [reply]
I don't use the x86 ,I want to build the src code through loong64 architecture to abtain the executable file like spectral-linux.

**Reply by P0lip** [reply]
Ah well, do you want the loong64 arch? I don't think Node.js distributes such binaries. 
Assuming you want to use [pkg](https://github.com/vercel/pkg-fetch), you'd need to build Node.js (they [added](https://github.com/nodejs/node/pull/41323) such a build flag) and then follow the suggestion [here](https://github.com/vercel/pkg/discussions/1367).

**Reply by lqywow** [reply]
I've built nodejs and obtain the executable file on loong64arch platform. So how can I fix the build command ? 

"Fetching base Node.js binaries to PKG_CACHE_PATH"  I don't want to fectch when I use the command : 

 yarn workspace @stoplight/spectral-cli build.nix

**Reply by lqywow** [reply]
Sory , I don't want to use pkg-fecth because I prepared the nodejs on loong64 platform ,how can I fix the command "yarn workspace @stoplight/spectral-cli build.nix". When I use this command , it will use pkg5.4.1 to fectch

**Reply by P0lip** [reply]
Assuming you've applied the patches as suggested [here](https://github.com/vercel/pkg/discussions/1367), you could simply insert that binary under `PKG_CACHE_PATH` so it can be resolved by https://github.com/vercel/pkg-fetch/blob/main/lib/places.ts#L40.

`export PKG_CACHE_PATH=./node-binaries yarn workspace @stoplight/spectral-cli build.nix`.
Just make sure to name them properly

**Reply by lqywow** [reply]
Yes， I add the binary file to PKG_CACHE_PATH,but pkg still fetch the file from remote.

This picture shows my PKG_CACHE_PATH and  binary files.
![f70e7457eccf39d94ed4a4588410253](https://user-images.githubusercontent.com/28998027/171080361-dde7099f-6102-459b-9af1-504216b35cef.jpg)

My pkg version is 5.4.7,when I input the command  "yarn workspace @stoplight/spectral-cli build.nix" it still use 5.4.1 and don't find the binary file .

![8a8952b96ca23a89c88a26e7911f02f](https://user-images.githubusercontent.com/28998027/171080082-92a0113c-9f6f-4edb-abbb-ecb707a337bc.jpg)

**Reply by P0lip** [reply]
could you send me the Node.js binary you built?
